### DISCUSSION TITLE
Placing Inner Block Before "Contact Information" in Checkout

### DISCUSSION THREAD

**Discussion by tarunparswani26** [woocommerce/woocommerce#44963] [WooCommerce, Blocks, and Site Editing]
Hi,

Do you have any suggestions on how I can insert my inner block before the `Contact Information` block (`woocommerce/checkout-contact-information-block`)? 

My intention is to make my block the first inner block within the checkout block. I have examined the source code but couldn't find a viable solution. 

If you have any alternative solutions or suggestions regarding this matter, I would greatly appreciate your input.

**Comment by nielslange** [comment]
Hello @tarunparswani26,

Thanks for reaching out. To better understand your support request, can you answer the following questions:

- What is the purpose of the inner block?
- Why is it important that it's an inner block placed before the `Contact Information` block?
- In which way shall your inner block interact with the checkout process?

**Reply by tarunparswani26** [reply]
Hi @nielslange,

Trying to create an inner block that will show the `available coupons` specific to a particular logged-in user. 

I believe it would be beneficial for customers if we place the same inner block, such as `checkout-express-payment-block`, in the same position as it appears in the non-block view.

I attempted to place the same block in the available slots within the checkout block, but it didn't seem to work as intended. Additionally, there were no suitable slots to insert it after the Add a coupon section, within the `checkout-order-summary-coupon-form-block` inner block.

Please do let me know if you need more input.
